### PR TITLE
Add integration for a separate symbol resolver in VSCode and fix various overlay bugs

### PR DESCRIPTION
**Pull request by phschaad**
- Addresses the following bugs:
  - [x] FLOPS overlay stops being drawn when nodes are collapsed and reopened
  - [x] Edge shading still gets drawn in some circumstances where the edge itself isn't visible anymore
  - [x] Subgraphs' symbol mapping is ignored
  - [x] Symbol definition popup causes the renderer to crash in some circumstances
  - [x] Symbol definition popup doesn't show for nodes when it should
  - [x] Symbol definition popup appears even if the symbols were all defined

### CODE REVIEW

File: overlay_manager.js [hunk 1]
```diff
@@ -449,67 +531,211 @@ class MemoryVolumeOverlay extends GenericSdfgOverlay {
...
+            this.symbol_resolver.symbol_value_map,
+            volume_values
+        );
+
+        switch (this.badness_scale_method) {
+            case 'mean':
+                this.badness_scale_center = math.mean(volume_values);
+                break;
+            case 'mode':
+                this.badness_scale_center = math.mode(volume_values);
```
Comment (tbennun) [inline overlay_manager.js:627]: There can be more than one mode in math.js!
https://mathjs.org/docs/reference/functions/mode.html

File: overlay_manager.js [hunk 2]
```diff
@@ -265,50 +285,107 @@ class StaticFlopsOverlay extends GenericSdfgOverlay {
...
+            this.symbol_resolver.symbol_value_map,
+            flops_values
+        );
+
+        switch (this.badness_scale_method) {
+            case 'mean':
+                this.badness_scale_center = math.mean(flops_values);
+                break;
+            case 'mode':
+                this.badness_scale_center = math.mode(flops_values);
```
Comment (tbennun) [inline overlay_manager.js:370]: same as comment below.

**Review by tbennun — CHANGES_REQUESTED** [review]
one change

File: overlay_manager.js (hunk 1 above)
Comment (phschaad) [reply overlay_manager.js]: Good catch, thank you!

**Review by tbennun — APPROVED** [review]

**Pull request merged**